### DISCUSSION TITLE
Examples with React Select

### DISCUSSION THREAD

**Discussion by hailwood** [houseform/houseform#59] [Ideas]
One library I always run into issues with is [react select](https://react-select.com/home) when used with form libraries.
For example with Formik I often find that value updates from react select are delayed and don't cause a rerender until something else changes (e.g. an additional focus/blur of the field).

One case that tends to trigger this pretty easily is two fields, where based off the value of the first field, the second react-select field may be emptied (think select country > select region type setup).

It would be great to have a few examples using react-select with house form that show the forms stored values being updated and validation being triggered correctly - 
- Standard single select
- Standard multi select
- Clearing value/setting value externally

// ref: https://twitter.com/HailwoodNZ/status/1637913443956326400

**Comment by crutchcorn** [comment]
Hi @hailwood, apologies for the delay - I've had a lot going on.

https://stackblitz.com/edit/houseform-v1-example-react-select

These are the examples of using HouseForm with React Select you asked for :) 

While I did not include the `clearing value`/`setting value externally` section, that's because it works exactly as it does in our HouseForm docs page  :D 

https://houseform.dev/guides/access-data-externally.html

You should have no problems whatsoever with getting the `ref` usage working as-intended.

Let me know if you have more questions about HouseForm! :)
